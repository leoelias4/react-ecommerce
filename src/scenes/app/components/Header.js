import React from "react";
import { NavLink } from "react-router-dom";
import { css, cx } from "react-emotion";

const styles = {
  header: css`
    background: #2296f3;
    height: 50px;
    box-shadow: 0px 1px 5px 0px rgba(0, 0, 0, 0.2),
      0px 2px 2px 0px rgba(0, 0, 0, 0.14), 0px 3px 1px -2px rgba(0, 0, 0, 0.12);
    position: fixed;
    width: 100%;
    z-index: 9;
    label: header;
  `,
  nav: css`
    max-width: 960px;
    height: inherit;
    margin: 0 auto;
    display: grid;
    grid-template-columns: 100px 150px auto 100px;
    grid-gap: 5px;
    align-items: center;
    label: nav;
    @media (max-width: 768px) {
      grid-template-columns: 1fr 1fr 1fr;
    }
  `,
  link: css`
    height: inherit;
    line-height: 50px;
    text-align: center;
    text-decoration: none;
    color: #fff;
    font-size: 1.3rem;
    background: #1c80d0;
    transition: 0.3s all ease;
    position: relative;
    overflow: hidden;
    label: link;
    &:hover {
      background: #1a6daf;
    }
  `,
  card: css`
    grid-column: 4;
    @media (max-width: 768px) {
      grid-column: 3;
    }
    label: card;
  `,
  badge: css`
    min-width: 20px;
    height: 20px;
    position: absolute;
    top: 5px;
    right: 8px;
    border-radius: 10px;
    background: #ffda26;
    font-size: 0.8rem;
    display: flex;
    justify-content: center;
    align-items: center;
    color: #1c80d0;
    transition: 0.3s all ease;
    label: badge;
  `,
  adding: css`
    background: #65ff26;
    transform: scale(1.2);
  `,
  activeLink: css`
    && {
      background: #1a6daf;
    }
  `
};

class Header extends React.Component {
  state = { adding: false };
  componentWillReceiveProps(props) {
    if (this.props.cartItems !== props.cartItems) {
      this.setState({
        adding: true
      });
      window.setTimeout(() => {
        this.setState({
          adding: false
        });
      }, 300);
    }
  }
  render() {
    const badgeClass = this.state.adding
      ? cx(styles.badge, styles.adding)
      : styles.badge;
    return (
      <header className={styles.header}>
        <nav className={styles.nav}>
          <NavLink
            activeClassName={styles.activeLink}
            exact
            className={styles.link}
            to="/"
          >
            Home
          </NavLink>
          <NavLink
            activeClassName={styles.activeLink}
            exact
            className={styles.link}
            to="/Top-5"
          >
            Top 5
          </NavLink>
          <NavLink
            activeClassName={styles.activeLink}
            exact
            className={cx(styles.card, styles.link)}
            to="/cart"
          >
            Cart
            {this.props.cartItems > 0 && (
              <span className={badgeClass}>{this.props.cartItems}</span>
            )}
          </NavLink>
        </nav>
      </header>
    );
  }
}

export default Header;