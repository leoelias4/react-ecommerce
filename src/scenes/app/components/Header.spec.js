import React from "react";
import sinon from "sinon";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Header from "./Header";
configure({ adapter: new Adapter() });

const setup = props => {
  const component = shallow(<Header />);

  return {
    component: component
  };
};

describe("Product component", () => {
  it("should render successfully", () => {
    const { component } = setup();
    expect(component).toMatchSnapshot();
  });

  it("should call component once", () => {
    const { component } = setup();
    const spy = sinon.spy(Header.prototype, "componentWillReceiveProps");
    expect(spy.calledOnce).toEqual(false);
    component.setProps({ cartItems: "1" });
    expect(spy.calledOnce).toEqual(true);
    expect(component.state().adding).toBe(true);
    setTimeout(() => expect(component.state().adding).toBe(false), 300);
  });
});
