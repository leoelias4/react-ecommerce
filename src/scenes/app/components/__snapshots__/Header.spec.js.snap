// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product component should render successfully 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <nav
        className="css-15lufbm-nav"
>
        <NavLink
                activeClassName="css-1wmldo3"
                aria-current="page"
                className="css-1uuek28-link"
                exact={true}
                to="/"
        >
                Home
        </NavLink>
        <NavLink
                activeClassName="css-1wmldo3"
                aria-current="page"
                className="css-1uuek28-link"
                exact={true}
                to="/Top-5"
        >
                Top 5
        </NavLink>
        <NavLink
                activeClassName="css-1wmldo3"
                aria-current="page"
                className="css-996yo3-card-link"
                exact={true}
                to="/cart"
        >
                Cart
        </NavLink>
</nav>,
      "className": "css-113lo4b-header",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="css-1wmldo3"
            aria-current="page"
            className="css-1uuek28-link"
            exact={true}
            to="/"
>
            Home
</NavLink>,
          <NavLink
            activeClassName="css-1wmldo3"
            aria-current="page"
            className="css-1uuek28-link"
            exact={true}
            to="/Top-5"
>
            Top 5
</NavLink>,
          <NavLink
            activeClassName="css-1wmldo3"
            aria-current="page"
            className="css-996yo3-card-link"
            exact={true}
            to="/cart"
>
            Cart
</NavLink>,
        ],
        "className": "css-15lufbm-nav",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "css-1wmldo3",
            "aria-current": "page",
            "children": "Home",
            "className": "css-1uuek28-link",
            "exact": true,
            "to": "/",
          },
          "ref": null,
          "rendered": "Home",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "css-1wmldo3",
            "aria-current": "page",
            "children": "Top 5",
            "className": "css-1uuek28-link",
            "exact": true,
            "to": "/Top-5",
          },
          "ref": null,
          "rendered": "Top 5",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "css-1wmldo3",
            "aria-current": "page",
            "children": Array [
              "Cart",
              false,
            ],
            "className": "css-996yo3-card-link",
            "exact": true,
            "to": "/cart",
          },
          "ref": null,
          "rendered": Array [
            "Cart",
            false,
          ],
          "type": [Function],
        },
      ],
      "type": "nav",
    },
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <nav
          className="css-15lufbm-nav"
>
          <NavLink
                    activeClassName="css-1wmldo3"
                    aria-current="page"
                    className="css-1uuek28-link"
                    exact={true}
                    to="/"
          >
                    Home
          </NavLink>
          <NavLink
                    activeClassName="css-1wmldo3"
                    aria-current="page"
                    className="css-1uuek28-link"
                    exact={true}
                    to="/Top-5"
          >
                    Top 5
          </NavLink>
          <NavLink
                    activeClassName="css-1wmldo3"
                    aria-current="page"
                    className="css-996yo3-card-link"
                    exact={true}
                    to="/cart"
          >
                    Cart
          </NavLink>
</nav>,
        "className": "css-113lo4b-header",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="css-1wmldo3"
              aria-current="page"
              className="css-1uuek28-link"
              exact={true}
              to="/"
>
              Home
</NavLink>,
            <NavLink
              activeClassName="css-1wmldo3"
              aria-current="page"
              className="css-1uuek28-link"
              exact={true}
              to="/Top-5"
>
              Top 5
</NavLink>,
            <NavLink
              activeClassName="css-1wmldo3"
              aria-current="page"
              className="css-996yo3-card-link"
              exact={true}
              to="/cart"
>
              Cart
</NavLink>,
          ],
          "className": "css-15lufbm-nav",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "css-1wmldo3",
              "aria-current": "page",
              "children": "Home",
              "className": "css-1uuek28-link",
              "exact": true,
              "to": "/",
            },
            "ref": null,
            "rendered": "Home",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "css-1wmldo3",
              "aria-current": "page",
              "children": "Top 5",
              "className": "css-1uuek28-link",
              "exact": true,
              "to": "/Top-5",
            },
            "ref": null,
            "rendered": "Top 5",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "css-1wmldo3",
              "aria-current": "page",
              "children": Array [
                "Cart",
                false,
              ],
              "className": "css-996yo3-card-link",
              "exact": true,
              "to": "/cart",
            },
            "ref": null,
            "rendered": Array [
              "Cart",
              false,
            ],
            "type": [Function],
          },
        ],
        "type": "nav",
      },
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
